### DISCUSSION TITLE
How does Azure IPAM handle Virtual WAN?

### DISCUSSION THREAD

**Discussion by joffstar** [Azure/ipam#128] [Q&A]
Hi!

I believe I saw somewhere, that IPAM now support VWAN? 
I would like to ask if there is any guide to this feature in the docs?

In our project, we are making reservations for the VWAN address spaces, but in the (older) version that we run, when we send the XPAM-REF-ID tag nothing happened.

So I am wondering, if it's just a matter to pull the latest version of the IPAM app, and then it's just a matter of tagging the network range on the VWAN?

Any guidance here would be awesome :)

**Comment by DCMattyG** [comment]
Hey there @jonprattLBG, it's true!

First off, my apologies I am working on some updated docs to better showcase this feature.

The elevator pitch is that prior to this, we really only looked at Virtual Networks, but Virtual Hubs in vWAN are essentially a kind of Virtual Network that take a CIDR block too, so now we expose both Virtual Networks and Virtual Hubs in the IPAM tool, and for the sake of assignment to Blocks, we treat them identically.

Now, in your question above, were you asking if it's possible to create a reservations, then use that reservation to create a vWAN Virtual Hub instead of a Virtual Network and apply the TAG to have it automatically recognized by the IPAM tool (and vicariously associated to the target Block)?

**Reply by joffstar** [reply]
Hi Matt
Well I was asking if there was a tagging based approach for the VWAN & IPAM ...

But what I think you just said was we can send the VWAN resourceID to the blocks/networks endpoint to register the network and the IPs that it consumed.

We do this already with all our networks, as per your previous advice on how to handle multiple addr space per vnet. We wrote two TF modules using Terracurl to call the IPAM API 1) register the reservation (vwan uses this to grab a CIDR. but leaves a hanging reservation) and 2) a IPAM 'post' module that sends back the networks to register and the reservations to delete and clean up ...  :)

**Reply by DCMattyG** [reply]
FYI @jonprattLBG, with PR #153 you can now use Reservation Tags on both Virtual Networks as well as Virtual WAN Hubs. Hope that helps, please let me know if you have any questions!
